### PR TITLE
UDS & CS Agreement Edit/Delete UI

### PR DESCRIPTION
**Pull request by achen2401**
https://www.pivotaltracker.com/story/show/180711183
Allow editing/removing of UDS and CS Agreement entries

### CODE REVIEW

**Review by ivan-c — APPROVED** [review]
Looks good!

File: patientsearch/api.py [hunk 1]
```diff
@@ -282,7 +281,7 @@ def update_resource_by_id(resource_type, resource_id):
     token = validate_auth()
 
     try:
-        resource = json.loads(request.headers.get("body", ""))
+        resource = request.get_json()
```
Comment (ivan-c) [inline patientsearch/api.py:284]: @pbugni @achen2401
After clicking **View**, I'm getting a `400` on shared testing when PUT is called to search/create a patient.

I don't see any differences in the request from the frontend to backend, so I suspect this change is responsible, but I'm not sure why (`Content-type: application/json` is correctly included)

```
Bad Request

The browser (or proxy) sent a request that this server could not understand.
```

image

**Review by ivan-c — CHANGES_REQUESTED** [review]
Think I found a bug on shared testing; not sure...

File: patientsearch/src/js/components/PatientListTable.js [hunk 2]
```diff
@@ -316,9 +316,9 @@ export default function PatientListTable() {
           method: "PUT",
           headers: {
             Accept: "application/json",
-            "Content-Type": "application/json",
-            body: rowData.resource ? JSON.stringify(rowData.resource) : null,
+            "Content-Type": "application/json"
           },
+          body: rowData.resource ? JSON.stringify(rowData.resource) : null
```
Comment (pbugni) [inline patientsearch/src/js/components/PatientListTable.js:321]: YES! we should never be naming a `body` header.  Recipe for easy confusion.  Sorry if we had a broken API expecting that!  Should never be the case.

File: patientsearch/api.py (hunk 1 above)
Comment (pbugni) [reply patientsearch/api.py:284]: This change is correct, we had old dysfunctional code there, from the initial implementation of this API as far as I can tell.

Should never be naming or requesting a **header** named **body** - the body of the request is correct, and where we should look for the attached JSON; just an implementation confusion.  The front end code has also been corrected to match (but isn't live where you saw this error on shared testing).

Comment (ivan-c) [reply patientsearch/api.py:284]: Thanks for the explanation! looks good on shared testing now

**Pull request merged**